python/ray/serve/http_proxy.py

```diff
@@ -202,18 +188,13 @@ async def __init__(self, host, port, instance_name=None):
         asyncio.get_event_loop().create_task(self.run())
 
     async def run(self):
-        sock = socket.socket()
-        sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
-        sock.bind((self.host, self.port))
-        sock.set_inheritable(True)
-
-        config = uvicorn.Config(self.app, lifespan="on", access_log=False)
+        config = uvicorn.Config(self.app, lifespan="off", access_log=False)
         server = uvicorn.Server(config=config)
         # TODO(edoakes): we need to override install_signal_handlers here
         # because the existing implementation fails if it isn't running in
         # the main thread and uvicorn doesn't expose a way to configure it.
         server.install_signal_handlers = lambda: None
-        await server.serve(sockets=[sock])
+        await server.serve()
```

Comment:
Nice changes in this file, I had a feeling it was more complicated than it needed to be :)